server/src/test/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDeciderTests.java

```diff
@@ -931,7 +931,7 @@ public void testCanRemainWithShardRelocatingAway() {
         ClusterState clusterState = ClusterState.builder(baseClusterState).routingTable(builder.build()).build();
         RoutingAllocation routingAllocation = new RoutingAllocation(
             null,
-            new RoutingNodes(clusterState),
+            new RoutingNodes(clusterState.routingTable(), clusterState.nodes()),
```

Comment:
Would `clusterState.getRoutingNodes()` not work here too (and in the other tests using the constructor)?

Reply:
Yes, literally the only reason I didn't use it was because that would instantiate that field on the `ClusterState` and thus change the behavior of the test. I can't see how that would be relevant here in practice, but I didn't want to make any changes to this in this test (could have just called that method before as well).